src/test/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpointTest.java

```diff
@@ -72,6 +89,56 @@ public void testParseNullUrlAndImageName() throws Exception {
         new DockerRegistryEndpoint(null, null).imageName(null);
     }
 
+    @Issue("JENKINS-48437")
+    @Test
+    public void testGetTokenForItem() throws IOException {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        MockAuthorizationStrategy auth = new MockAuthorizationStrategy()
+                .grant(Jenkins.READ).everywhere().to("alice")
+                .grant(Computer.BUILD).everywhere().to("alice")
+                .grant(Item.CONFIGURE).everywhere().to("alice");
+        j.jenkins.setAuthorizationStrategy(auth);
+
+        String globalCredentialsId = "global-creds";
+        IdCredentials credentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL,
+                globalCredentialsId, "test-global-creds", "user", "password");
+        CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), credentials);
+
+        FreeStyleProject item = j.createFreeStyleProject("testGetToken");
+        SecurityContextHolder.getContext().setAuthentication(User.get("alice").impersonate());
+
+        DockerRegistryToken token = new DockerRegistryEndpoint("https://index.docker.io/v1/", globalCredentialsId).getToken(item);
+        Assert.assertNotNull(token);
+        Assert.assertEquals("user", token.getEmail());
+        Assert.assertEquals(Base64.encodeBase64String("user:password".getBytes(Charsets.UTF_8)), token.getToken());
+    }
+
+    @Issue("JENKINS-48437")
+    @Test
+    public void testGetTokenForRun() throws IOException {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        MockAuthorizationStrategy auth = new MockAuthorizationStrategy()
+                .grant(Jenkins.READ).everywhere().to("alice")
+                .grant(Computer.BUILD).everywhere().to("alice")
+                .grant(Item.CONFIGURE).everywhere().to("alice");
+        j.jenkins.setAuthorizationStrategy(auth);
+
+        String globalCredentialsId = "global-creds";
+        IdCredentials credentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL,
+                globalCredentialsId, "test-global-creds", "user", "password");
+        CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), credentials);
+
+        FreeStyleProject item = j.createFreeStyleProject("testGetToken");
```

Comment:
BTW `@Rule JenkinsRule` starts a new instance for each test case, so there is no need to pass a job name argument.